contracts/ThalesRoyale/ThalesRoyalePrivateRoom.sol

```diff
@@ -247,324 +336,395 @@ contract ThalesRoyalePrivateRoom is Owned, Pausable {
                 _populateRewardForRoom(_roomNumber, totalPlayersInARoomInARound[_roomNumber][currentRound]);
                 emit SplitBetweenLoosers(_roomNumber, currentRound, totalPlayersInARoomInARound[_roomNumber][currentRound]);
-            } else{ 
+            } else {
                 // there is min 1 winner
+                numberOfWinneres = winningPositionsPerRound;
                 _populateRewardForRoom(_roomNumber, winningPositionsPerRound);
             }
 
             roomEndTime[_roomNumber] = block.timestamp;
             // first close previous round then royale
             emit RoundClosedInRoom(_roomNumber, currentRound, roundResultPerRoom[_roomNumber][currentRound]);
-            emit RoyaleFinishedForRoom(_roomNumber);
+            emit RoyaleFinishedForRoom(_roomNumber, numberOfWinneres, rewardPerWinnerPerRoom[_roomNumber]);
         } else {
+            // need to reverse result because of isPlayerAliveInASpecificRoom() in positioning a new round so the play can continue
+            if (winningPositionsPerRound == 0 && gameTypeInRoom[_roomNumber] == GameType.LAST_MAN_STANDING) {
+                isReversedPositioningInRoom[_roomNumber] = true;
+            }
+
             roundStartTimeInRoom[_roomNumber] = block.timestamp;
             roundEndTimeInRoom[_roomNumber] = roundStartTimeInRoom[_roomNumber] + roundLengthInRoom[_roomNumber];
             emit RoundClosedInRoom(_roomNumber, currentRound, roundResultPerRoom[_roomNumber][currentRound]);
         }
-        
     }
 
-    function claimRewardForRoom(uint _roomNumber) external onlyWinners(_roomNumber){
+    function claimRewardForRoom(uint _roomNumber) external onlyWinners(_roomNumber) {
         require(rewardPerRoom[_roomNumber] > 0, "Reward must be set");
-        require(rewardPerPlayerPerRoom[_roomNumber] > 0, "Reward per player must be more then zero");
-        require(rewardCollectedPerRoom[_roomNumber][msg.sender] == false, "Player already collected reward");
-        require(block.timestamp < (roomEndTime[_roomNumber] + climeTimePerRoom[_roomNumber]), "Time for reward claiming expired");
-
-        // get balance 
-        uint balance = rewardToken.balanceOf(address(this));
-        
-        if (balance != 0){
-
-            // set collected -> true
-            rewardCollectedPerRoom[_roomNumber][msg.sender] = true;
-            
-            // transfering rewardPerPlayer
-            rewardToken.transfer(msg.sender, rewardPerPlayerPerRoom[_roomNumber]);
-
-            // emit event
-            emit RewardClaimed(_roomNumber, msg.sender, rewardPerPlayerPerRoom[_roomNumber]);
-        }
+        require(!rewardCollectedPerRoom[_roomNumber][msg.sender], "Player already collected reward");
+
+        // set collected -> true
+        rewardCollectedPerRoom[_roomNumber][msg.sender] = true;
+        unclaimedRewardPerRoom[_roomNumber] = unclaimedRewardPerRoom[_roomNumber].sub(rewardPerWinnerPerRoom[_roomNumber]);
```

Comment:
the comment from above [ use add() or sub() ]  is implemented here. 
I think is good to be everywhere